File: app/src/main/java/com/android/kotemanagement/room/database/KoteDatabase.java
package com.android.kotemanagement.room.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.android.kotemanagement.room.dao.AdminDao;
import com.android.kotemanagement.room.dao.IssueWeaponsDao;
import com.android.kotemanagement.room.dao.RecordsDao; // Import the new DAO
import com.android.kotemanagement.room.dao.SoldiersDao;
import com.android.kotemanagement.room.entities.Admin;
import com.android.kotemanagement.room.entities.IssueWeapons;
import com.android.kotemanagement.room.entities.Records; // Import the new entity
import com.android.kotemanagement.room.entities.Soldiers;

@Database(entities = {Soldiers.class, IssueWeapons.class, Records.class, Admin.class}, version = 4)
public abstract class KoteDatabase extends RoomDatabase {

    abstract public SoldiersDao getSoldiersDao();
    abstract public IssueWeaponsDao getIssueWeaponsDao();
    abstract public RecordsDao getRecordsDao();
    abstract public AdminDao getAdminDao();

    private static volatile KoteDatabase instance = null;

    public static KoteDatabase getDatabaseInstance(Context context) {
        synchronized (KoteDatabase.class) {
            if (instance == null) {
                instance = Room.databaseBuilder(context.getApplicationContext(), KoteDatabase.class, "kote_database")
                        .fallbackToDestructiveMigration()
                        .build();
            }
            return instance;
        }
    }
}

File: app/src/main/java/com/android/kotemanagement/roomdatabase/database/KoteDatabase.kt
package com.android.kotemanagement.roomdatabase.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.sqlite.db.SupportSQLiteDatabase
import com.android.kotemanagement.roomdatabase.dao.SoldiersDAO
import com.android.kotemanagement.roomdatabase.entities.Soldiers
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch

@Database(entities = [Soldiers::class], version = 1)
abstract class KoteDatabase : RoomDatabase() {

    abstract fun getSoldiersDao() : SoldiersDAO

    //Singleton making
    companion object {
        @Volatile
        private var instance : KoteDatabase? = null

        fun getDatabase(context: Context) : KoteDatabase{
            synchronized(this) {
                if(instance == null) {
                    instance = Room.databaseBuilder(context.applicationContext, KoteDatabase::class.java,
                        "kote_database").build()
                }
                return instance as KoteDatabase
            }
        }
    }



}
